src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala

```diff
@@ -577,16 +574,21 @@ trait ClusterComponent extends LeoComponent {
     } yield (cluster, instance, error, label, extension, image, scopes)
   }
 
-  private def fullClusterQueryByUniqueKey(googleProject: GoogleProject,
+  def fullClusterQueryByUniqueKey(googleProject: GoogleProject,
                                           clusterName: ClusterName,
-                                          destroyedDateOpt: Option[Instant]) = {
+                                          destroyedDateOpt: Option[Instant] = None): Query[(ClusterTable, Rep[Option[InstanceTable]], Rep[Option[ClusterErrorTable]], Rep[Option[LabelTable]], Rep[Option[ExtensionTable]], Rep[Option[ClusterImageTable]], Rep[Option[ScopeTable]]), (ClusterRecord, Option[InstanceRecord], Option[ClusterErrorRecord], Option[LabelRecord], Option[ExtensionRecord], Option[ClusterImageRecord], Option[ScopeRecord]), Seq] = {
     val destroyedDate = destroyedDateOpt.getOrElse(dummyDate)
 
-    fullClusterQuery
-      .filter { _._1.googleProject === googleProject.value }
-      .filter { _._1.clusterName === clusterName.value }
-      .filter { _._1.destroyedDate === Timestamp.from(destroyedDate) }
-      .result
+    for {
+      ((((((cluster, instance), error), label), extension), image), scopes) <-
+        clusterQuery filter (_.googleProject === googleProject.value) filter (_.clusterName === clusterName.value) filter (_.destroyedDate === Timestamp.from(destroyedDate)) joinLeft
```

Comment:
as far as i can tell, the key here is applying the filters on the initial `clusterQuery` rather than on the entire result of the `fullClusterQuery`. doing it afterwards resulted in large derived tables which slowed down the query considerably. the sql that this generates is equally horrific but the explain plan is much better

Reply:
Before:
<img width="1257" alt="Screen Shot 2019-07-02 at 11 13 07 AM" src="https://user-images.githubusercontent.com/7257391/60532887-55e76400-9ccc-11e9-96f0-7c30f9822141.png">

After:
<img width="1292" alt="Screen Shot 2019-07-02 at 11 12 47 AM" src="https://user-images.githubusercontent.com/7257391/60532902-597aeb00-9ccc-11e9-8b7a-6a9a79955853.png">